Added a custom labels visualization example

jaimedp opened the pull request:
Added a custom labels visualization example that puts labels at the end of the chart or pinned to the outside of the bars.

mmrj commented on examples/showcase/bar/bar-export-2/demo.details:
```diff
@@ -0,0 +1,7 @@
+---
+ name: Bar Chart with custom data labels
+ description: Here we've created a custom visualization that puts the data labels at the end of the chart align with each bar, or at the end of the bar.
```
nit: align -> aligned
question: which side is the "end of the chart"? since it's a bar chart, do you always mean the right?

jaimedp replied:
Updated the text slightly.

narenranjit commented:
Finally got around to looking at this. A few questions:

- http://jsfiddle.net/o0nz3f6k/3/ on small widths the text overlaps with the bar. In that specific case, making the bars smaller to make the labels fit may be the right solution, but I'm not sure if that's generic enough / if that's even possible as a custom visualization. Thoughts?

- Can you replace arrow function with regular to make it consistent with the rest of the examples? (also has a couple of minor lint issues)

- General n00b question - each time my data changes I'm doing `.setData(foo).customLegend(foo)`.  However each time I do this, this adds a new `#customLegend` node to the svg and also duplicates contents with each node (see screenshot).
<img width="992" alt="screen shot 2017-11-22 at 1 09 42 pm" src="https://user-images.githubusercontent.com/830274/33149989-8db68330-cf86-11e7-80d6-102267693603.png">

Am I doing something wrong?

narenranjit commented:
It looks like the extra `g`s are probably triggered by my resize handler
`                window.addEventListener('resize', function() { _this.contourInstance.resize().render(); });
` but I'm not sure why?

jaimedp commented:
hey... I removed the arrow functions and fixed the linting issues... as for the extra copies of the legends, you are not supposed to call `customLegend(foo)` again when you call `setData`... `setData` automatically propagates the new version of the data to all visualizations... 

Have you tried just calling `setData(foo).render()`? if it doesn't work there may be a different problem.

As for the labels overlapping the bars on small widths, I think there could be a few options. 1) move the labels outside of the chart "frame" so they lay outside of the axis, that would prevent the overlapping but it may look strange. 2) change the color of the labels so they are actually readable when they overlap or 3) hide the label and only show it on over or something like that? Not sure if any of the solutions is generic enough and it may be better to treat it on a case by case.

narenranjit commented:
This works, jsfiddle for the record http://jsfiddle.net/o0nz3f6k/5/
